docs/user-guide/spark-user-guide.md

```diff
@@ -0,0 +1,159 @@
+
+## Program assets
+
+The cohorting capabilities (aka [quality-measure-and-cohort-service](https://github.com/Alvearie/quality-measure-and-cohort-service/)) are delivered as a public open source project under the Alvearie organization in Github. The module of interest for users consuming the Apache Spark function is  [cohort-evaluator-spark](https://github.com/Alvearie/quality-measure-and-cohort-service/tree/cql-spark-evaluator). Provided inside that project is a sample [Dockerfile](https://github.com/Alvearie/quality-measure-and-cohort-service/blob/cql-spark-evaluator/cohort-evaluator-spark/Dockerfile) that can be built and published to the appropriate place in your execution environment. No publicly published version of the docker image exists at this time.
+
+## Program Execution
+
+The Spark application is built on top of Apache Spark 3.1.2 running on top of Hadoop 3.2 or later. Users wanting to run the application should first [download](https://spark.apache.org/downloads.html) and extract the appropriate Spark distribution to their local system and set a SPARK_HOME environment property that points to the location of the extracted files. Windows users will also need to do some additional configuration of Hadoop for Windows by downloading pre-compiled DLLs from https://github.com/cdarlint/winutils/tree/master/hadoop-3.2.2/ and setting HADOOP_HOME environment variable to location of the downloaded files.
+
+The entry point to the Spark application is the `com.ibm.cohort.cql.spark.SparkCqlEvaluator` class. The typical mode of invocation is to use the appropriate spark-submit script from SPARK_HOME\bin to execute the application. The simplest invocation for a Windows user will look like this...
+
+```
+%SPARK_HOME%\bin\spark-submit.cmd --name cohort-evaluator-spark --class com.ibm.cohort.cql.spark.SparkCqlEvaluator local:/opt/spark/jars/cohort-evaluator-spark/target/cohort-evaluator-spark-1.0.2-SNAPSHOT.jar @/path/to/program-args.txt
+```
+
+In this example, the `@/path/to/program-args.txt` is a convenience option that allows users to store their program arguments in a file that can be reused across multiple program executions. Each line of the target file is a separate positional parameter. The option name and option value (for those options that have a value) should be on separate lines. Make sure to avoid trailing whitespace as that can confuse the argument parser. Alternatively, you can just pass all the options (e.g. -c /path/to/cql -m /path/to/modelinfo.xml -d /path/to/context-definitions.json -j /path/to/cql-jobs.json ...) directly in the spark-submit execution.
```

Comment:
May be need to provide example of output from usage statement to show the list of program-args that are available.